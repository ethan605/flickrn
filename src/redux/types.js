export const PREFIXES = {
  FLICKR: 'flickr/',
  SHARED: 'shared/',
};

export const FLICKR = {
  CLEAR_PHOTO_INFO: 'flickr/CLEAR_PHOTO_INFO',
  GET_PHOTO_INFO: 'flickr/GET_PHOTO_INFO',
  SEARCH_PHOTOS: 'flickr/SEARCH_PHOTOS',
};

export const SHARED = {
};
