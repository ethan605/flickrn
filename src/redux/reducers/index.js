import photoInfo from './photoInfo';
import photos from './photos';

export { photoInfo, photos };
