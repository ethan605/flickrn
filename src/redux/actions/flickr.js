import _ from 'lodash';
import { createAction } from 'redux-actions';
import { FLICKR } from 'src/redux/types';

// Constants
import { API_CONFIGS } from 'src/constants';

const { flickr: { apiKey } } = require('src/data/configs.json');
const { PAGINATION } = API_CONFIGS;

const DEFAULT_PARAMS = {
  api_key: apiKey,
  format: 'json',
  nojsoncallback: 1,
  per_page: PAGINATION,
};

export const clearPhotoInfo = createAction(FLICKR.CLEAR_PHOTO_INFO);

export function getPhotoInfo({ id, secret }) {
  const action = createAction(FLICKR.GET_PHOTO_INFO);

  return dispatch => {
    const request = {
      params: {
        ...DEFAULT_PARAMS,
        secret,
        method: 'flickr.photos.getInfo',
        photo_id: id,
      },
    };

    return dispatch(action({ request }));
  };
}

export function searchPhotos({ keyword, reloading }) {
  const action = createAction(FLICKR.SEARCH_PHOTOS);

  return (dispatch, getState) => {
    const { photos: { currentPage } } = getState();

    const method = _.isEmpty(keyword) ? 'getRecent' : 'search';
    
    const request = {
      params: {
        ...DEFAULT_PARAMS,
        method: `flickr.photos.${method}`,
        page: reloading ? 1 : currentPage + 1,
        text: keyword,
      },
    };

    return dispatch(action({ request }));
  };
}
