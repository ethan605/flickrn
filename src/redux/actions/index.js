import * as flickr from './flickr';

/* eslint-disable import/prefer-default-export */

export { flickr };
