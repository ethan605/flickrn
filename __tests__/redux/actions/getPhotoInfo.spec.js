// Helpers
import { DEFAULT_PARAMS, buildAxiosActions, mockAxios } from '__tests__/helpers/mockApiRequests';
import buildMockStore from '__tests__/helpers/buildMockStore';

// Redux
import { flickr } from 'src/redux/actions';

describe('redux/actions/getPhotoInfo', () => {
  const store = buildMockStore({ authorized: true });

  const id = 123;
  const secret = 'random_secret_xxx';
  const mockSuccess = require('__tests__/fixtures/getPhotoInfo/success.json');
  const mockError = require('__tests__/fixtures/getPhotoInfo/error.json');

  const requestAction = {
    type: 'flickr/GET_PHOTO_INFO',
    payload: {
      request: {
        params: {
          ...DEFAULT_PARAMS,
          secret,
          method: 'flickr.photos.getInfo',
          photo_id: id,
        },
      },
    },
  };

  const { photo } = mockSuccess;
  const { errorAction, successAction } = buildAxiosActions(requestAction, { photo }, mockError);

  /* eslint-disable newline-per-chained-call */
  mockAxios
    .onGet('').replyOnce(200, mockSuccess)
    .onGet('').replyOnce(200, mockError);
  /* eslint-enable newline-per-chained-call */

  beforeEach(() => store.clearActions());

  test('success', async () => {
    await store.dispatch(flickr.getPhotoInfo({ id, secret }));
    const actions = store.getActions();
    expect(actions.length).toEqual(2);
    expect(actions).toEqual([requestAction, successAction]);
    expect(actions).toMatchSnapshot();
  });

  test('error', async () => {
    await store.dispatch(flickr.getPhotoInfo({ id, secret }));
    const actions = store.getActions();
    expect(actions.length).toEqual(2);
    expect(actions).toEqual([requestAction, errorAction]);
    expect(actions).toMatchSnapshot();
  });
});
