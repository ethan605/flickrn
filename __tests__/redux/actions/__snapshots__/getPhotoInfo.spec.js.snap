// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redux/actions/getPhotoInfo error 1`] = `
Array [
  Object {
    "payload": Object {
      "request": Object {
        "params": Object {
          "api_key": "92ab2a894f0effc28f920362ff52fc2d",
          "format": "json",
          "method": "flickr.photos.getInfo",
          "nojsoncallback": 1,
          "per_page": 20,
          "photo_id": 123,
          "secret": "random_secret_xxx",
        },
      },
    },
    "type": "flickr/GET_PHOTO_INFO",
  },
  Object {
    "error": Object {
      "message": "Random flickr.photos.getInfo error",
      "response": Object {
        "data": Object {
          "code": 123,
          "message": "Random flickr.photos.getInfo error",
          "stat": "fail",
        },
      },
    },
    "meta": Object {
      "previousAction": Object {
        "payload": Object {
          "request": Object {
            "params": Object {
              "api_key": "92ab2a894f0effc28f920362ff52fc2d",
              "format": "json",
              "method": "flickr.photos.getInfo",
              "nojsoncallback": 1,
              "per_page": 20,
              "photo_id": 123,
              "secret": "random_secret_xxx",
            },
          },
        },
        "type": "flickr/GET_PHOTO_INFO",
      },
    },
    "type": "flickr/GET_PHOTO_INFO:ERROR",
  },
]
`;

exports[`redux/actions/getPhotoInfo success 1`] = `
Array [
  Object {
    "payload": Object {
      "request": Object {
        "params": Object {
          "api_key": "92ab2a894f0effc28f920362ff52fc2d",
          "format": "json",
          "method": "flickr.photos.getInfo",
          "nojsoncallback": 1,
          "per_page": 20,
          "photo_id": 123,
          "secret": "random_secret_xxx",
        },
      },
    },
    "type": "flickr/GET_PHOTO_INFO",
  },
  Object {
    "meta": Object {
      "previousAction": Object {
        "payload": Object {
          "request": Object {
            "params": Object {
              "api_key": "92ab2a894f0effc28f920362ff52fc2d",
              "format": "json",
              "method": "flickr.photos.getInfo",
              "nojsoncallback": 1,
              "per_page": 20,
              "photo_id": 123,
              "secret": "random_secret_xxx",
            },
          },
        },
        "type": "flickr/GET_PHOTO_INFO",
      },
    },
    "payload": Object {
      "photo": Object {
        "comments": Object {
          "_content": "0",
        },
        "dates": Object {
          "lastupdate": "1521094692",
          "posted": "1521085336",
          "taken": "2018-03-15 06:28:31",
          "takengranularity": 0,
          "takenunknown": "1",
        },
        "dateuploaded": "1521085336",
        "description": Object {
          "_content": "Hair : #Tableau Vivant  \\\\\\\\ Silvana hair <a href=\\"https://maps.secondlife.com/secondlife/8 8/108/194/1085\\" rel=\\"nofollow\\">Collabor88</a>

Outfit :   #New in <a href=\\"http://maps.secondlife.com/secondlife/kustom9/124/120/22\\" rel=\\"nofollow\\">kustom9 </a>
                 #=Zenith= Girl Heart Uniforms gacha 
                 =Zenith=Mobile neck case gacha
                 =Zenith=Leather Star Boots with Sock gacha 
    
backpack :  =Zenith=Cat Backpack (all Color) -RARE gacha 

 Cat : #JIAN Chubby Cat Bird Bus

Dogs : #+Half-Deer+  Corgi Puppy - Walking Companion RARE
            +Half-Deer+    Corgi Puppy - Cuddle Companion RARE
",
        },
        "editability": Object {
          "canaddmeta": 0,
          "cancomment": 0,
        },
        "farm": 5,
        "id": "40107995774",
        "isfavorite": 0,
        "license": "0",
        "media": "photo",
        "notes": Object {
          "note": Array [],
        },
        "originalformat": "png",
        "originalsecret": "0e602e4261",
        "owner": Object {
          "iconfarm": 5,
          "iconserver": "4798",
          "location": "",
          "nsid": "152508136@N06",
          "path_alias": null,
          "realname": "jodi H",
          "username": "jody2015SL",
        },
        "people": Object {
          "haspeople": 0,
        },
        "publiceditability": Object {
          "canaddmeta": 0,
          "cancomment": 1,
        },
        "rotation": 0,
        "safety_level": "0",
        "secret": "7ed5905c16",
        "server": "4778",
        "tags": Object {
          "tag": Array [],
        },
        "title": Object {
          "_content": "üê±üê∂üê∂ üíñ",
        },
        "urls": Object {
          "url": Array [
            Object {
              "_content": "https://www.flickr.com/photos/152508136@N06/40107995774/",
              "type": "photopage",
            },
          ],
        },
        "usage": Object {
          "canblog": 0,
          "candownload": 1,
          "canprint": 0,
          "canshare": 1,
        },
        "views": "244",
        "visibility": Object {
          "isfamily": 0,
          "isfriend": 0,
          "ispublic": 1,
        },
      },
    },
    "type": "flickr/GET_PHOTO_INFO:SUCCESS",
  },
]
`;
